opentelemetry-etw-logs/src/logs/exporter.rs

```diff
@@ -0,0 +1,347 @@
+use async_trait::async_trait;
+use eventheader::{FieldFormat, Level, Opcode};
+use eventheader_dynamic::EventBuilder;
+use std::borrow::Cow;
+use std::collections::HashMap;
+use std::fmt::Debug;
+use std::sync::Arc;
+
+use opentelemetry::{logs::AnyValue, logs::Severity, Key};
+use std::{cell::RefCell, str, time::SystemTime};
+
+/// Provider group associated with the user_events exporter
+pub type ProviderGroup = Option<Cow<'static, str>>;
+
+thread_local! { static EBW: RefCell<EventBuilder> = RefCell::new(EventBuilder::new());}
+
+/// Exporter config
+#[derive(Debug)]
+pub struct ExporterConfig {
+    /// keyword associated with user_events name
+    /// These should be mapped to logger_name as of now.
+    pub keywords_map: HashMap<String, u64>,
+    /// default keyword if map is not defined.
+    pub default_keyword: u64,
+}
+
+impl Default for ExporterConfig {
+    fn default() -> Self {
+        ExporterConfig {
+            keywords_map: HashMap::new(),
+            default_keyword: 1,
+        }
+    }
+}
+
+impl ExporterConfig {
+    pub(crate) fn get_log_keyword(&self, name: &str) -> Option<u64> {
+        self.keywords_map.get(name).copied()
+    }
+
+    pub(crate) fn get_log_keyword_or_default(&self, name: &str) -> Option<u64> {
+        if self.keywords_map.is_empty() {
+            Some(self.default_keyword)
+        } else {
+            self.get_log_keyword(name)
+        }
+    }
+}
+pub(crate) struct UserEventsExporter {
```

Comment:
Should it be named as `ETWExporter` instead of `UserEventsExporter` ? As [user_events](https://docs.kernel.org/trace/user_events.html) refers to the trace-events in user space for Linux.

Reply:
Yes, I just forked the UserEvents project, but going to replace with ETW code.